src/store.js

```diff
@@ -282,6 +282,13 @@ function installModule (store, rootState, path, module, hot) {
     const parentState = getNestedState(rootState, path.slice(0, -1))
     const moduleName = path[path.length - 1]
     store._withCommit(() => {
+      if (process.env.NODE_ENV !== 'production') {
+        if (moduleName in parentState) {
+          console.warn(
+            `[vuex] state field "${moduleName}" was overridden by a module with the same name`
```

Comment:
@simplesmiler what do u think about including the module name in the latter part as well? `[vuex] state field "${moduleName}" was overridden by a module with the same name ("${moduleName}")`